src/Dictionaries/RangeHashedDictionary.cpp

```diff
@@ -178,10 +177,76 @@ ColumnPtr RangeHashedDictionary::getColumn(
     return result;
 }
 
-ColumnUInt8::Ptr RangeHashedDictionary::hasKeys(const Columns &, const DataTypes &) const
+ColumnUInt8::Ptr RangeHashedDictionary::hasKeys(const Columns & key_columns, const DataTypes & key_types) const
 {
-    throw Exception(ErrorCodes::NOT_IMPLEMENTED,
-        "Has not supported", getDictionaryID().getNameForLogs());
+    auto range_storage_column = key_columns[1];
+    ColumnWithTypeAndName column_to_cast = {range_storage_column->convertToFullColumnIfConst(), key_types[1], ""};
+
+    auto range_column_storage_type = std::make_shared<DataTypeInt64>();
+    auto range_column_updated = castColumnAccurate(column_to_cast, range_column_storage_type);
```

Comment:
Why cast?

Reply:
RangeHashed dictionary contains id and numeric representable range with size lower than 8 bytes, user fetch is based on id and this range, we cast range to Int64, to process it.